src/test/java/sirius/web/service/JsonOutputSpec.groovy

```diff
@@ -0,0 +1,33 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.web.service
+
+import groovy.json.JsonSlurper
+import sirius.kernel.BaseSpecification
+import sirius.kernel.xml.Attribute
+import sirius.web.services.JSONStructuredOutput
+
+class JsonOutputSpec extends BaseSpecification {
+    def "json output uses attributes"() {
+        given:
+        OutputStream os = new ByteArrayOutputStream();
+        JSONStructuredOutput out = new JSONStructuredOutput(os, null, "UTF8");
+        when:
+        out.beginResult("test");
+        out.beginObject("1", Attribute.set("a", "b"))
+        out.endObject()
+        out.endResult();
+        then:
+        compareJsonString(os.toString(), """{"1":{"a":"b"}}""");
+    }
+
+    private def compareJsonString(String str1, String str2) {
```

Comment:
not needed?

Reply:
Depends on https://github.com/scireum/sirius-web/pull/747#discussion_r482841806